### PR TITLE
Share map with clustered features

### LINKED ISSUE #910: Share map link to marked object
**Describe the bug**
When an object is marked the share map link does not pan/zoom to that object.

**To Reproduce**
Mark an object, press share map control and open the url i new window.

**Expected behavior**
The map should pan/zoom to the object that was marked before getting the link.

**Additional context**
Dont find any documentation for this feature.

### PR DESCRIPTION
**Pull request by mulfvik**
Fixes #910

### CODE REVIEW

**Comment by jokd** [comment]
Tested and the map still doesn't pan to the feature. Tested on Edge Chromium with the origo-cities geojson as cluster layer.

**Comment by mulfvik** [comment]
> Tested and the map still doesn't pan to the feature. Tested on Edge Chromium with the origo-cities geojson as cluster layer.

Only tested with WFS so my bad, looking in to it.

**Comment by mulfvik** [comment]
Found one more bug, when infowindow is infowindow the zoom to object wont work at all. No id in url is present

File: src/viewer.js [hunk 1]
```diff
@@ -411,17 +411,21 @@ const Viewer = function Viewer(targetOption, options = {}) {
...
-            if (type === 'WFS') {
+            if (clusterSource && type === 'WFS') {
+              feature = clusterSource.getFeatureById(featureId);
+            } else if (type === 'WFS') {
               feature = layer.getSource().getFeatureById(featureId);
             } else {
               const id = featureId.split('.')[1];
               let origin = layer.getSource();
               feature = origin.getFeatureById(id);
+              feature = layer.getSource().getFeatureById(featureId);
```
Comment (jokd) [inline src/viewer.js:428]: I think this row just messes things up.

Comment (mulfvik) [reply src/viewer.js:428]: Refactored this and came up with that topojson has problems with this feature
             `if (type === 'WFS' && clusterSource) {
              feature = clusterSource.getFeatureById(featureId);
              console.log(feature)

            } else if (type === 'WFS') {
              feature = layer.getSource().getFeatureById(featureId);
              console.log(feature)

            } else if (type === 'GEOJSON' && clusterSource) {
              feature = clusterSource.getFeatureById(id);
              console.log(feature)

            } else if (type === 'GEOJSON') {
              feature = layer.getSource().getFeatureById(id);
              console.log(feature)

            } else if (type === 'TOPOJSON' && clusterSource) {
              feature = clusterSource.getFeatureById(id);
              console.log(feature)

            } else if (type === 'TOPOJSON') {
              feature = layer.getSource().getFeatureById(id);
              console.log(feature)
            }`

Comment (jokd) [reply src/viewer.js:428]: Is that a working solution?

Comment (mulfvik) [reply src/viewer.js:428]: This is an example where WFS and GEOJSON is returning a feature but TOPOJSON returning null. Not sure why, the getFeatureById is pretty straight forward..

Comment (jokd) [reply src/viewer.js:428]: could it be your layer source?

Comment (mulfvik) [reply src/viewer.js:428]: Yes, Im confused where in the prototype the feature is found..

Comment (jokd) [reply src/viewer.js:428]: I tried the socken.json topojson file and that file seems to lack IDs. The id in the permalink is perhaps the OL id.

Comment (mulfvik) [reply src/viewer.js:428]: If i share Övik (id 16) in geojson. Permalink is referencing origo-cities.7. I **thought** getFeatureById was straight forward, I was wrong.
Testing on fme converted origo-cities.geojson -> origo-cities.json and in topojson i get id 94

Comment (mulfvik) [reply src/viewer.js:428]: So, topojson object id should not be nested inside properties. There is nothing wrong with the code as far as I can se.

**Pull request merged**